src/main/java/software/aws/neptune/common/gremlindatamodel/MetadataCache.java

```diff
@@ -68,6 +68,21 @@ public static void updateCache(final String endpoint, final int port, final bool
         }
     }
 
+    /**
+     * Function to update the cache of the metadata.
+     * @param gremlinConnectionProperties GremlinConnectionProperties to use.
+     * @throws SQLException
+     */
+    public static void updateGremlinCache(final GremlinConnectionProperties gremlinConnectionProperties)
+            throws SQLException {
+        final String endpoint = gremlinConnectionProperties.getContactPoint();
+        synchronized (LOCK) {
+            if (!GREMLIN_SCHEMAS.containsKey(endpoint)) {
+                GREMLIN_SCHEMAS.put(endpoint, SchemaHelperGremlinDataModel.getGremlinGraphSchema(gremlinConnectionProperties));
+            }
+        }
```

Comment:
Would consider unrolling `SchemaHelperGremlinDataModel.getGremlinGraphSchema(gremlinConnectionProperties)` function since it's just a call to a static function. Since there's a lock here, it's nice to see explicitly what actions are under the lock if/when possible